### ISSUE TITLE
The async pragma doesn't work properly with a nested proc inside a generic function

### ISSUE THREAD

**Issue by zah** [nim-lang/Nim#10818]
The following example fails to compile with the the error

 ```
example.nim(9, 11) Error: undeclared identifier: 'await'
```

### Example
```nim
import
  asyncdispatch

type
  AsyncProc = proc(): Future[void]

proc makeCallback[T](v: T): AsyncProc =
  result = proc() {.async.} =
    await sleepAsync(100)
  
let c = makeCallback(10)
```

The problem manifests only when the `makeCallback` proc is generic. At the moment, it's possible to work-around the issue by manually simulating the low-level mechanics of `await`:

```nim
proc makeCallback[T](v: T): AsyncProc =
  result = proc() {.async.} =
    var future = sleepAsync(100)
    yield future
    if future.failed: ...
```

### Additional Information
```
$ nim -v
Nim Compiler Version 0.19.9 [Linux: amd64]
git hash: b9f8528db675927aacc6443ae3a363711bb41945
active boot switches: -d:release
```

**Comment by mratsim** [comment]
Sounds like another case of https://github.com/nim-lang/Nim/issues/8677 - generics early symbol resolution issue

**Comment by ringabout** [comment]
same issue as #15804
And fixed by #15820
